contracts/src/v0.8/automation/ExecutionPrevention.sol

```diff
@@ -23,4 +23,24 @@ abstract contract ExecutionPrevention {
     _preventExecution();
     _;
   }
+
+  /**
+   * @notice modifier that allows it to be simulated via eth_call by checking
+   * that the sender is the zero address.
+   */
+  modifier cannotExecuteExcept(address allowedOrigin) {
+    _preventExecutionExcept(allowedOrigin);
+    _;
+  }
+
+  /**
+   * @notice method that allows it to be simulated via eth_call by checking that
+   * the sender is the zero address.
+   */
+  function _preventExecutionExcept(address allowedOrigin) internal view {
+    // solhint-disable-next-line avoid-tx-origin
+    if (tx.origin != allowedOrigin) {
+      revert OnlySimulatedBackend();
+    }
+  }
```

Comment:
I'd be ok if we removed this contract from our inheritance chain, and defined our own `cannotExecute` modifier in `AutomationBase` - the code would probably be cleaner that way.